src/main/scala/org/apache/spark/sql/delta/storage/BaseExternalLogStore.scala

```
@@ -90,21 +90,53 @@ abstract class BaseExternalLogStore(sparkConf: SparkConf, hadoopConf: Configurat
    * Method try to rewrite TransactionLog from temporary path if it not exists.
    * Method returns completed [[LogEntryMetadata]]
    */
-
-   private def tryFixTransactionLog(fs: FileSystem, entry: LogEntryMetadata): LogEntryMetadata = {
+  private def tryFixTransactionLog(fs: FileSystem, entry: LogEntryMetadata): LogEntryMetadata = {
     logDebug(s"Try to fix: ${entry.path}")
-    val tempPath = entry.tempPath.get
     try {
-      writeActions(fs, entry.path, read(tempPath).iterator)
-      val completedEntry = entry.complete()
-      writeCache(fs, completedEntry, overwrite = true)
-      deleteFile(fs, tempPath)
-      completedEntry
+      writeLogTransaction(fs, entry)
     } catch {
-      case e: FileNotFoundException =>
-        logWarning(s"ignoring $e while fixing (already done?)")
-        entry
+      case _: FileSystemException => print("Failed to fix transaction, ignore...")
```

let's use logWarning here.
And I think I see some dangerous edge case here. You are ignoring all errors of writeLogTransaction (and you are catching all errors there), so it is possible that file is not renamed at all and you will mark it as complete in dynamodb below, right? I think the solution is to ignore only this FileNotFoundException (as it may occur only if other reader already renamed file) and propagate others

---

I agree with @mrk-its on this one. This will be actually rather common scenario, when 2 separate processes will be trying to do the same operation on the file.